src/resolver/lockfile.cc

```diff
@@ -234,6 +234,15 @@ to_json( nlohmann::json & jto, const LockedInputRaw & raw )
 }
 
 
+/* -------------------------------------------------------------------------- */
+
+std::ostream &
+operator<<( std::ostream & oss, const LockedInputRaw & raw )
+{
+  return oss << nlohmann::json( raw ).dump();
```

Comment:
Nonblocking
```suggestion
  return oss << nlohmann::json( raw ).dump(4);
```

Reply:
What does the 4 do?

Reply:
I added the `<<` op for `EXPECT_EQ`.
The test suite is going to format those errors like this if we add newlines and indents:
```
Expectation failed: ( ( input1 ) == ( input2 ) ). Got '{
  "foo": 1,
  "bar": 2
}' != '{
  "quux": 3
}'
```

If you prefer that I'm happy to apply the change; just wanted to highlight the behavior.

Reply:
I was thinking for some of the large JSON blobs having it formatted would be nicer, e.g. an entire lockfile. I agree it's worse for the first and last line. I don't have super strong feelings